dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ExecutorServiceImpl.java

```diff
@@ -1125,6 +1125,15 @@ public int createComplementDependentCommand(List<Schedule> schedules, Command co
                         CronUtils.getMaxCycle(schedules.get(0).getCrontab()), dependentCommand.getWorkerGroup());
         dependentCommand.setTaskDependType(TaskDependType.TASK_POST);
         for (DependentProcessDefinition dependentProcessDefinition : dependentProcessDefinitionList) {
+
+            long processDefinitionCode = dependentProcessDefinition.getProcessDefinitionCode();
+            ProcessDefinition processDefinition = processService.findProcessDefinitionByCode(processDefinitionCode);
```

Comment:
Querying data in a long loop will result in poor performance. We should avoid this.